tiny_dnn/nodes.h

```diff
@@ -43,24 +43,32 @@ namespace cereal {
 
 template <typename Archive>
 void save(Archive & ar, const std::vector<tiny_dnn::layerptr_t>& v) {
+#ifndef CNN_NO_SERIALIZATION
     ar(cereal::make_size_tag((cereal::size_type)v.size()));
     for (auto n : v) {
         tiny_dnn::layer::save_layer(ar, *n);
     }
+#else
+    throw tiny_dnn::nn_error("TinyDNN was not build with Serialization support");
```

Comment:
I think moving `cereal` folder into `third_party` would be good.
And what about adding `tiny_dnn/util/serialization` folder and moving `deserialization_helper.h`, `serialization_helper.h`, `serialization_layer_list.h` files into the folder.

`external` means we can separate existing `internal` functions that use `cereal` library into a separated file. Please take a look at the explanation on this page. http://uscilab.github.io/cereal/serialization_functions.html